QuEST/src/QuEST.c

```diff
@@ -1250,6 +1250,12 @@ PauliHamil createPauliHamilFromFile(char* fn) {
 	return h;
 }
 
+DiagonalOp createDiagonalOpFromPauliHamilFile(char* fn, QuESTEnv env, qreal* offset){
+
+	PauliHamil hamil = createPauliHamilFromFile(fn);
+	return createDiagonalOpFromPauliHamil(hamil, env, offset);
```

Comment:
must not call other API functions (defined in `QuEST.c`) from within an API function (muddies stack trace on exception, and duplicates validation)